python/ray/dag/compiled_dag_node.py

```diff
@@ -668,20 +663,21 @@ def _get_node_id(self):
             ] = worker_fn.options(concurrency_group="_ray_system").remote(
                 do_exec_tasks,
                 executable_tasks,
-                type_hints=list(set(self._type_hints)),
             )
 
-        # Wrapper function for inputs provided to dag.execute().
         input_task = self.idx_to_task[self.input_task_idx]
-        self.input_wrapper_fn = input_task.output_wrapper_fn
+        # Register custom serializers for inputs provided to dag.execute().
+        input_task.dag_node.type_hint.register_custom_serializer()
+
         self.dag_input_channel = input_task.output_channel
-        _do_register_custom_serializers(self, list(set(self._type_hints)))
 
         self.dag_output_channels = []
         for output in self.idx_to_task[self.output_task_idx].args:
             assert isinstance(output, DAGNode)
             output_idx = self.dag_node_to_idx[output]
             self.dag_output_channels.append(self.idx_to_task[output_idx].output_channel)
+            # Register custom serializers for DAG outputs.
+            output.type_hint.register_custom_serializer()
```

Comment:
QQ: does it have to be called like this? Isn't it possible to just make it a part of type hint constructor?

Reply:
Good idea, let me do that.

Reply:
Hmm actually going to hold off on this. We may not always want to register the serializer, i.e. if the value is just getting passed through the driver to another actor. Also, we may need to wait until after the constructor before we can define the serializer.